### PR TITLE
aaLength template

### PR DESCRIPTION
**Pull request by somzzz**
DMDchanges: https://github.com/somzzz/dmd/commit/ff6b28d3fab8a229380bd2b3ecd6462cb654920e

### CODE REVIEW

File: src/rt/aaA.d [hunk 1]
```diff
@@ -38,11 +38,11 @@ struct AA
...
     alias impl this;
+}
 
-    private @property bool empty() const pure nothrow @nogc
-    {
-        return impl is null || !impl.length;
-    }
+private @property bool empty(in Impl* aa) pure nothrow @nogc
+{
+    return aa is null || !aa.length;
```
Comment (schveiguy) [inline src/rt/aaA.d]: What is the reason for making this a function instead of a member?

File: src/rt/aaA.d [hunk 2]
```diff
@@ -347,11 +347,14 @@ private T max(T)(T a, T b) pure nothrow @nogc
 //------------------------------------------------------------------------------
 
 /// Determine number of entries in associative array.
+
```
Comment (schveiguy) [inline src/rt/aaA.d]: stray extra space here

File: src/rt/aaA.d [hunk 3]
```diff
@@ -347,11 +347,14 @@ private T max(T)(T a, T b) pure nothrow @nogc
...
 extern (C) size_t _aaLen(in AA aa) pure nothrow @nogc
 {
     return aa ? aa.length : 0;
 }
 
+
+
```
Comment (schveiguy) [inline src/rt/aaA.d]: more extra spaces

File: src/rt/aaA.d [hunk 4]
```diff
@@ -642,10 +645,12 @@ extern (C) int _aaEqual(in TypeInfo tiRaw, in AA aa1, in AA aa2)
     return true;
 }
 
+
+
```
Comment (schveiguy) [inline src/rt/aaA.d]: Spaces

File: src/rt/aaA.d [hunk 5]
```diff
@@ -477,7 +480,7 @@ extern (C) void _aaClear(AA aa) pure nothrow
 /// Rehash AA
 extern (C) void* _aaRehash(AA* paa, in TypeInfo keyti) pure nothrow
 {
-    if (!paa.empty)
+    if (!(*paa).empty)
```
Comment (schveiguy) [inline src/rt/aaA.d]: If you do keep `empty` as a non-member, I'd rather see it take a `const(AA*)` instead of this rewrite.

**Comment by schveiguy** [comment]
ping @somzzz is this still being pursued?

**Comment by RazvanN7** [comment]
Closing as this will be taken care of by @teodutu during SAoC

**Pull request closed without merging**